### ISSUE TITLE
--download option for md5sum

### ISSUE THREAD

**Issue by aicioara** [rclone/rclone#3923]
#### What is your current rclone version (output from `rclone version`)?

```
rclone v1.50.2
- os/arch: linux/amd64
- go version: go1.13.4
```

#### What problem are you are trying to solve?

The md5sum that rclone stores with the object is not guaranteed to be the md5sum of the actual object for multipart files (see https://github.com/rclone/rclone/issues/523). This means that the output of the following command is not guaranteed to be correct

```
rclone md5sum remote://path/to/file
```

This then cascades to `rclone check` as well.

#### How do you think rclone should be changed to solve that?

Rclone should add a `--download` flag for the md5sum command (possibly for hashsum as well) which downloads the file and returns the md5sum.

`--download` is a suggestion based on the `rclone check` command, but other options should also work.

**Comment by ncw** [comment]
That isn't a bad suggestion.. Do you want to work on it?

You could use `rclone cat remote:path/to/file | md5sum -` as a work-around.

**Comment by ivandeex** [comment]
this feature was merged by #4803 and released with 1.54
